resources/shaders/ambient_occlusion.hlsl

```diff
@@ -14,11 +14,16 @@ struct AOHitInfo
 cbuffer CBData : register(b0)
 {
 	float4x4 inv_vp;
+	float4x4 inv_view;
 
 	float bias;
 	float radius;
 	float power;
+	float max_distance;
+	float frame_idx;
 	unsigned int sample_count;
+
+	float2 padding;
```

Comment:
Yeah, cbuffers need vectors to be aligned 16-byte. This will add empty padding after sample_count (8 bytes) and then add the padding (8 bytes) and then add padding to that (8 more bytes). The correct version would be to swap float2 padding before the frame_idx. To keep track of this, you can structure the code like following:
```cpp
float x;
float y;
float z;
uint x;

float2 ab;
uint c, d;
```
Every 16-bytes gets a line of whitespace. Every 16-bytes can only have 1 vector, which has to be at the start of it.

Reply:
Looking into this!

Reply:
dabbed on it!!!